torch/_higher_order_ops/while_loop.py

```diff
@@ -204,13 +218,72 @@ def _is_boolean_scalar_tensor(pred):
 )
 
 
+def _create_unbacked_symint() -> torch.SymInt:
+    from torch._subclasses.fake_tensor import FakeTensorMode
+    from torch.fx.experimental.symbolic_shapes import ShapeEnv
+
+    # It's possible that there is no active fake tenosr mode, if we're tracing with
+    # "real" mode. so we create a new one.
```

Comment:
You mean make_fx with tracing_mode = real?

Reply:
yeah